cypress/integration/unit_tests/polygon_draw_test.js

```diff
@@ -178,37 +178,81 @@ describe('Unit test for ShapeData', () => {
   });
 
   it('Shows calculating before update finishes', () => {
-    const div = document.createElement('div');
-    document.body.appendChild(div);
-    const map = new google.maps.Map(div, {center: {lat: 0, lng: 0}, zoom: 1});
-    const polygon = new google.maps.Polygon({
-      map: map,
-      paths: [{lat: 0, lng: 0}, {lat: 1, lng: 1}, {lat: 0, lng: 1}],
-    });
     const event = new Event('overlaycomplete');
-    event.overlay = polygon;
+    cy.document()
+        .then((document) => {
+          const div = document.createElement('div');
+          document.body.appendChild(div);
+          const map =
+              new google.maps.Map(div, {center: {lat: 0, lng: 0}, zoom: 1});
+          event.overlay = new google.maps.Polygon({
+            map: map,
+            paths: [{lat: 0, lng: 0}, {lat: 1, lng: 1}, {lat: 0, lng: 1}],
+          });
+          return cy.window().then(
+              (win) => setUpPolygonDrawing(map, Promise.resolve(), win));
+        })
+        .then((drawingManager) => {
+          google.maps.event.trigger(drawingManager, 'overlaycomplete', event);
+          updatePromise.setPromise(event.resultPromise);
+        });
     firebaseCollection.add = () => Promise.resolve({id: 'id'});
     const updatePromise = new SettablePromise();
-    cy.wrap(setUpPolygonDrawing(map, Promise.resolve()))
+    cy.get('.popup-calculated-data').contains('calculating');
```

Comment:
Does this work because it automatically waits for all the stuff in cy.document()? We maybe want to pick a strategy between document.getElementBy*, jquery selectors and cy selectors.

Reply:
Yes, cy commands automatically wait for each other. I'm trying to use cy as much as possible: I'm only using the raw document for writing, and always use cy for reading.